src/hotspot/share/gc/shared/taskqueue.hpp

```diff
@@ -561,8 +561,10 @@ class ObjArrayTask
 
 class PartialArrayState;
 
-// Discriminated union over oop*, narrowOop*, and PartialArrayState.
+// Discriminated union over oop/oop*, narrowOop*, and PartialArrayState.
 // Uses a low tag in the associated pointer to identify the category.
+// Oop/oop* are overloaded using the same tag because they can not appear at the
```

Comment:
The reason why "they can not appear at the same time" is that one is for young-gc and the other full-gc, right? If so, maybe state that explicitly.

Reply:
As far as Parallel, `PromotionManager` uses `ScannerTask` to wrap around `*oop` and `CompactionManager` to wrap around `oop`, which is true. But I don't think it has such correlation. 

Probably should phase as `ScannerTask` wraps `*oop` or `oop` using the same tag, because they cannot be mixed in the same queue.

Reply:
> But I don't think it has to have such correlation.

I believe this is due to the choice of diff gc-algorithms. Young-gc uses scavenging (copying), while full-gc uses mark-compact (lisp-2). The former performs "pointer remap" in the same phase as forwarding.

(Ofc, this is just for comment/doc. Feel free to keep as is, or revise, or maybe other reviewers have sth to share.)